### PR TITLE
Update parseq-tracevis UI and dependencies

### PR DESCRIPTION
**Pull request by zackthehuman**
The trace visualizer UI hasn't been updated in many years. This lack of updates is catching up with us, since the JavaScript ecosystem moves fast. This PR introduces JavaScript tooling updates to get the trace visualizer at least somewhat up to date.

The high-level list of changes are:

- Update `nodejs` to 20.8.1
- Update `npm` to 8.15.0
- Update `bootstrap` to v4 (not the latest, but still updated)
- Replace `PhantomJS` with `Puppeteer` for integration testing
- Updated the look and feel of the visualizer slightly due to changes in how `bootstrap` works in v4 and partly because the existing UI needs a facelift. There is still a lot of room for improvement, but I kept the minimalist approach while trying to improve some of the usability and visual alignment of things.
  - The different view selectors have icons and correct mouse cursors.
  - The JSON code is now monospaced.
  - Labels and buttons are visually aligned and use the same vertical rhythm.
  - The waterfall view's container has optional horizontal scrollbars rather than the entire page.
  - The colors have been tweaked slightly due to changes in `bootstrap`.
  - HTML markup has been updated to remove some of the inlined JavaScript.
 
## Screenshot
parseq-trace-viewer

### CODE REVIEW

File: subprojects/parseq-tracevis/build.gradle [hunk 1]
```diff
@@ -11,10 +12,10 @@ apply plugin: 'com.github.node-gradle.node'
 
 node {
   // Version of node to use.
-  version = '6.11.2'
+  version = '20.8.1'
```
Comment (aliciatang) [inline subprojects/parseq-tracevis/build.gradle:15]: Wow 14 major versions!  Very nice!

File: subprojects/parseq-tracevis/package.json [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
     "cp-data": "1.1.3",
-    "jshint": "~2.1.11",
-    "istanbul": "~0.1.44",
-    "mocha": "1.2.x",
-    "phantomjs": "^2.1.7",
+    "istanbul": "^0.4.5",
+    "jshint": "^2.13.6",
+    "mocha": "^10.2.0",
+    "puppeteer": "^21.3.8",
     "uglify-js": "2.6.0"
```
Comment (aliciatang) [inline subprojects/parseq-tracevis/package.json:13]: nit: not up for the latest version? `3.17.4`?

File: subprojects/parseq-tracevis/trace.html [hunk 3]
```diff
@@ -27,64 +27,157 @@
...
+  fill: #0056b3;
+}
+
+.nav-pills .nav-link.active svg {
+  fill: var(--white);
+}
+
+.icon {
+  width: 20px;
+  height: 20px;
```
Comment (aliciatang) [inline subprojects/parseq-tracevis/trace.html:53]: nit: not using `rem` for this one?

File: subprojects/parseq-tracevis/trace.html [hunk 4]
```diff
@@ -27,64 +27,157 @@
...
+  </symbol>
+</svg>
```
Comment (aliciatang) [inline subprojects/parseq-tracevis/trace.html:103]: nit: in a separate svg file for image?

**Review by aliciatang — APPROVED** [review]
Thanks for maintaining it!

File: subprojects/parseq-tracevis/trace.html (hunk 4 above)
Comment (zackthehuman) [reply subprojects/parseq-tracevis/trace.html:103]: I inlined it so there will be no delay in loading the file, and the SVG wasn't too big. I'd prefer to not load additional external files in this case.

File: subprojects/parseq-tracevis/build.gradle (hunk 1 above)
Comment (zackthehuman) [reply subprojects/parseq-tracevis/build.gradle:15]: I was surprised that nothing totally broke after this many majors :)

File: subprojects/parseq-tracevis/trace.html (hunk 3 above)
Comment (zackthehuman) [reply subprojects/parseq-tracevis/trace.html:53]: Hmm, let me see if I can revise this.

**Review by junchuanwang — APPROVED** [review]

**Comment by zackthehuman** [comment]
@aliciatang Thanks for the feedback. I will address the nits in a future update. I'd like to land this first. When I have cycles, I want to make more updates to the JS code and other parts of the UI.

**Pull request merged**